dockerfiles/Dockerfile.build

```diff
@@ -1,25 +1,11 @@
 FROM golang:1.7.4-alpine
 
-# A container for building InfraKit
-
 RUN apk add --update git make
+RUN go get github.com/rancher/trash
 
-ENV CGO_ENABLED=0
-ENV GOPATH=/go
-ENV PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
-
-# Development tools
-RUN go get github.com/rancher/trash \
-           github.com/golang/lint/golint \
-           github.com/golang/mock/gomock \
-           github.com/golang/mock/mockgen
-
-# The project sources
-ADD ./ /go/src/github.com/docker/infrakit
 WORKDIR /go/src/github.com/docker/infrakit
-
 VOLUME [ "/go/src/github.com/docker/infrakit/build" ]
+CMD make build-binaries
 
+COPY . ./
```

Comment:
+ `COPY` is recommended over `ADD` (https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/)
+ Copying to `./` copies to the `WORKINGDIR`, so no issue, just some duplication removed.